src/context/OracleClientContext.tsx

```diff
@@ -1,38 +1,51 @@
 import React, { useEffect, useState } from "react";
-import { events, oracle, client } from "helpers/oracleClient";
+import { events, oracle, clients, mapObject } from "helpers/oracleClient";
 
-export type State = {
-  state: oracle.types.state.State;
+type States = {
+  [key in oracle.types.state.OracleType]?: oracle.types.state.State;
```

Comment:
each oracle type now has independent state, so we add the new key in the heirarchy.